image_builder/image_builder/job_type.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+from collections import defaultdict
+from dataclasses import dataclass
+from pathlib import Path
+
+import backoff
+
+from racetrack_client.log.context_error import wrap_context
+from racetrack_client.utils.semver import SemanticVersion
+from racetrack_commons.plugin.core import PluginCore
+from racetrack_commons.plugin.engine import PluginEngine
+
+
+@dataclass
+class JobTypeVersion:
```

Comment:
I have ambigous feeling for the name JobType**Version** - looking at the class, it looks more like JobType. Don't see reason to emphasize that it stores version... it stores other fields too. It's visible also in the names of the functions below - `load_job_type`, `gather_job_types`. Food for thought :)

Reply:
You got me. I've noticed I'm using it interchangeably `job_type: JobTypeVersion`, so it definitely makes sense to call it `JobType`